src/Components/Components/src/Routing/Router.cs

```diff
@@ -56,6 +58,12 @@ public class Router : IComponent, IHandleAfterRender, IDisposable
         /// </summary>
         [Parameter] public RenderFragment<RouteData> Found { get; set; }
 
+        /// <summary>
+        /// Gets or sets a handler that should be called before navigating to a new page. Return `true`
+        /// if the route's component should be rendered following invocation. Return `false` otherwise.
+        /// </summary>
+        [Parameter] public Func<string, bool> OnNavigate { get; set; }
```

Comment:
I don't want to block on this for p7, but we could conceive of making this behave like a filter:

```C#
Action<NavigationContext> OnNavigate;

NavigationContext
{
   string BaseRelativePath { get; }
   bool Cancel { get; }
}
```

This saves you the trouble of having to return a value if all you needed to was inspect the route. I'm not convinced it's necessary, just some musings

Reply:
Tracked this in https://github.com/dotnet/aspnetcore/issues/23499.